okhttp/src/main/java/okhttp3/internal/http/CallServerInterceptor.kt

```diff
@@ -56,7 +56,12 @@ class CallServerInterceptor(private val forWebSocket: Boolean) : Interceptor {
         } else {
           // Write the request body if the "Expect: 100-continue" expectation was met.
           val bufferedRequestBody = exchange.createRequestBody(request, false).buffer()
-          requestBody.writeTo(bufferedRequestBody)
+          try {
+            requestBody.writeTo(bufferedRequestBody)
+          } catch (ioe: IOException) {
```

Comment:
I think this is the right fix but the wrong place. I’d prefer for `RetryAndFollowUpInterceptor` to do this because that’s where all the other exceptions are caught and handled.

Reply:
CallServerInterceptor allowed for understanding that a request body was being written, so the HTTP/1.1 stream is well and truly borked.

RetryAndFollowUpInterceptor gets an IOException, in this case exactly the exception that RequestBody throws, so it seems tough to make any decision based on it.

Does calling detachWithViolence from transmitter.exchangeDoneDueToException achieve the same goal?

Reply:
I think it’s probably correct to use detachWithViolence() whenever get an IOException from the chain of network interceptors. It’ll make sure everything is released and put back to normal.

Reply:
(which is to say... the current commit is perfect)